subsys/bluetooth/mesh/rpl.c

```diff
@@ -121,226 +64,70 @@ bool bt_mesh_rpl_check(struct bt_mesh_net_rx *rx,
 		return false;
 	}
 
-	for (i = 0; i < ARRAY_SIZE(replay_list); i++) {
-		struct bt_mesh_rpl *rpl = &replay_list[i];
+	struct bt_mesh_rpl *rpl = NULL;
 
-		/* Empty slot */
-		if (!rpl->src) {
-			if (match) {
-				*match = rpl;
-			} else {
-				bt_mesh_rpl_update(rpl, rx);
-			}
+	if (bt_mesh_rpl_get(rx->ctx.addr, &rpl)) {
+		BT_ERR("RPL is full!");
+		return true;
+	}
 
-			return false;
+	if (!rpl) {
```

Comment:
I see your point. I was considering this check back and forth for that exact reason. It really should not happen that `rpl = NULL` here, but I thought it might be wise since this is a pointer and there might be multiple implementations for the body of the `bt_mesh_rpl_get`.

Reply:
If I don't miss anything, `bt_mesh_rpl_get` could just return pointer to `struct bt_mesh_rpl_get` instead of int.